registry/service.go

```diff
@@ -52,11 +53,12 @@ func (s *Service) Auth(job *engine.Job) engine.Status {
 			if err != nil {
 				return job.Error(err)
 			}
+			secure = endpoint.secure
```

Comment:
I think this entire section can be:

index, err := ResolveIndexInfo(job, authConfig.ServerAddress)
if err != nil {
	return job.Error(err)
}
endpoint, err := NewEndpoint(index)
if err != nil {
	return job.Error(err)
}
secure = endpoint.secure
authConfig.ServerAddress = endpoint.String()


ServerAddress should never be "", and the Official index shouldn't need to be special-cased